src/native/corehost/host_runtime_contract.h

```diff
@@ -23,13 +23,37 @@
 #define HOST_PROPERTY_PLATFORM_RESOURCE_ROOTS "PLATFORM_RESOURCE_ROOTS"
 #define HOST_PROPERTY_TRUSTED_PLATFORM_ASSEMBLIES "TRUSTED_PLATFORM_ASSEMBLIES"
 
+struct trusted_platform_assemblies
+{
+	uint32_t assembly_count;
+	char** basenames; /* Foo.dll */
+	char** assembly_filepaths; /* /blah/blah/blah/Foo.dll */
+};
+
+struct probing_lookup_paths 
+{
+	uint32_t dir_count;
+	char** dirs;
+};
+
+struct probing_path_properties
+{
+    trusted_platform_assemblies trusted_platform_assemblies;
+    probing_lookup_paths native_dll_search_directories;
+    probing_lookup_paths platform_resource_roots;
+};
+
 struct host_runtime_contract
 {
     size_t size;
 
     // Context for the contract. Pass to functions taking a contract context.
     void* context;
 
+    char* entry_assembly;
```

Comment:
Nit: Should any new things be added to the end of this structure so that it grows in backward compatible way?